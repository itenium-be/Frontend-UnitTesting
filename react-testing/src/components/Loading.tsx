import { useState } from 'react';
import { WiredCheckbox, WiredDivider, WiredInput, WiredSpinner } from './wrappers/WiredElements';


export function Loading() {
  const [running, setRunning] = useState(true)
  const [color, setColor] = useState('red')

  return (
    <>
      Loading...
      <br />
      <WiredSpinner spinning={running} duration={1000} style={{color}} />


      <WiredDivider style={{marginTop: 20}} />


      <div style={{paddingTop: 32}}>
        <h3>Change Properties</h3>
        <WiredCheckbox checked={running} onChange={() => setRunning(!running)} /> Running?
        <br />

        Duration:
        <WiredInput placeholder="1500" style={{width: 100}} />

        <br />
        Color:
        <WiredInput placeholder="Pick a web color" value={color} onChange={evnt => setColor(evnt.target.value)} style={{width: 100}} />
      </div>
    </>
  )
}
