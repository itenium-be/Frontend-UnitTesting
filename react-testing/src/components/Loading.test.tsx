import renderer from 'react-test-renderer';
import { Loading } from './Loading';

// Jest Snapshot Testing
// You may want to use Snapshot Testing sparsely
//
// https://jestjs.io/docs/snapshot-testing

// Update snapshot when the component changes
// jest --updateSnapshot
// npm test -- -u

// toMatchSnapshot() vs toMatchInlineSnapshot()
// --> Work with external files in __snapshots__
// --> vs put the snapshots inline in the same file

describe('Loading screen', () => {
  it('renders "Loading... by default', () => {
    const tree = renderer
      .create(<Loading />)
      .toJSON();

    expect(tree).toMatchSnapshot();

    // If you do not care about certain parts of the render:
    // expect(user).toMatchSnapshot({
    //   createdAt: expect.any(Date),
    //   id: expect.any(Number),
    // });

    // or replace manually before toMatchSnapshot:
    // tree = tree.replace(/\w{32}/, 'ID')
  })

  it('can change the spinner color', () => {
    // TODO: should probably have a wired color picker of sorts
    const component = renderer.create(<Loading />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = component.root;
    const colorInput = testInstance.findByProps({placeholder: "Pick a web color"})

    renderer.act(() => {
      colorInput.props.onChange({target: {value: 'green'}})
    })
    expect(colorInput.props.value).toEqual('green');

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('can set a custom "Loading..." message', () => {})
})
