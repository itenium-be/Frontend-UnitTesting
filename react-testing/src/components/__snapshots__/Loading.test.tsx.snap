// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading screen can change the spinner color 1`] = `
Array [
  "Loading...",
  <br />,
  <wired-spinner
    style={
      Object {
        "color": "red",
      }
    }
  />,
  <wired-divider
    style={
      Object {
        "marginTop": 20,
      }
    }
  />,
  <div
    style={
      Object {
        "paddingTop": 32,
      }
    }
  >
    <h3>
      Change Properties
    </h3>
    <wired-checkbox />
     Running?
    <br />
    Duration:
    <wired-input
      style={
        Object {
          "width": 100,
        }
      }
    />
    <br />
    Color:
    <wired-input
      style={
        Object {
          "width": 100,
        }
      }
    />
  </div>,
]
`;

exports[`Loading screen can change the spinner color 2`] = `
Array [
  "Loading...",
  <br />,
  <wired-spinner
    style={
      Object {
        "color": "green",
      }
    }
  />,
  <wired-divider
    style={
      Object {
        "marginTop": 20,
      }
    }
  />,
  <div
    style={
      Object {
        "paddingTop": 32,
      }
    }
  >
    <h3>
      Change Properties
    </h3>
    <wired-checkbox />
     Running?
    <br />
    Duration:
    <wired-input
      style={
        Object {
          "width": 100,
        }
      }
    />
    <br />
    Color:
    <wired-input
      style={
        Object {
          "width": 100,
        }
      }
    />
  </div>,
]
`;

exports[`Loading screen renders "Loading... by default 1`] = `
Array [
  "Loading...",
  <br />,
  <wired-spinner
    style={
      Object {
        "color": "red",
      }
    }
  />,
  <wired-divider
    style={
      Object {
        "marginTop": 20,
      }
    }
  />,
  <div
    style={
      Object {
        "paddingTop": 32,
      }
    }
  >
    <h3>
      Change Properties
    </h3>
    <wired-checkbox />
     Running?
    <br />
    Duration:
    <wired-input
      style={
        Object {
          "width": 100,
        }
      }
    />
    <br />
    Color:
    <wired-input
      style={
        Object {
          "width": 100,
        }
      }
    />
  </div>,
]
`;
